pkg/vere/mars.c

```diff
@@ -0,0 +1,272 @@
+/* worker/mars.c
+**
+**  the main loop of a mars process.
+*/
+#include "c3.h"
+#include "noun.h"
+#include "types.h"
+#include "vere.h"
+#include "db/lmdb.h"
+#include <mars.h>
+#include <stdio.h>
+
+/* _mars_step_trace(): initialize or rotate trace file.
+*/
+static void
+_mars_step_trace(const c3_c* dir_c)
+{
+  if ( u3C.wag_w & u3o_trace ) {
+    if ( u3_Host.tra_u.con_w == 0  && u3_Host.tra_u.fun_w == 0 ) {
+      u3t_trace_open(dir_c);
+    }
+    else if ( u3_Host.tra_u.con_w >= 100000 ) {
+      u3t_trace_close();
+      u3t_trace_open(dir_c);
+    }
+  }
+}
```

Comment:
The trace module was refactored upstream, so this function should change to match `_serf_step_trace()` (the only difference being that the pier path is passed in).